internal/nodes/avi_model_l7_hostname_shard.go

```diff
@@ -167,43 +180,68 @@ func (o *AviObjectGraph) BuildPoolPGPolicyForDedicatedVS(vsNode []*AviVsNode, na
 			// first, and that is going to mess up the ordering. Hence creating a pool with a different name here. The previous pool will become stale in the process and will get deleted.
 			// An AKO reboot would be required to clean up any stale pools if left behind.
 			poolName = poolName + "--" + lib.PoolNameSuffixForHttpPolToPool
-			httpPGPath.Pool = poolName
+			if lib.CheckObjectNameLength(poolName, lib.Pool) {
+				isPoolNameLenExceedAviLimit = true
+			}
+			if !isPoolNameLenExceedAviLimit {
+				// Add only when pool name is < 255
+				httpPGPath.Pool = poolName
+			}
 			utils.AviLog.Infof("key: %s, msg: using pool name: %s instead of poolgroups for http policy set", key, poolName)
 		} else {
 			pgName := lib.GetSniPGName(ingName, namespace, hostname, obj.Path, infraSettingName, vsNode[0].Dedicated)
-			//var pgfound bool
+			if lib.CheckObjectNameLength(pgName, lib.PG) {
+				isPGNameLenExceedAviLimit = true
+			}
 			pgNode, pgfound = localPGList[pgName]
 			if !pgfound {
 				pgNode = &AviPoolGroupNode{Name: pgName, Tenant: lib.GetTenant()}
 			}
 			localPGList[pgName] = pgNode
-			httpPGPath.PoolGroup = pgNode.Name
+			if !isPGNameLenExceedAviLimit {
+				httpPGPath.PoolGroup = pgNode.Name
+			}
 			pgNode.AviMarkers = lib.PopulatePGNodeMarkers(namespace, hostname, infraSettingName, []string{ingName}, []string{obj.Path})
 		}
 		var storedHosts []string
 		storedHosts = append(storedHosts, hostname)
 		poolNode := buildPoolNode(key, poolName, ingName, namespace, priorityLabel, hostname, infraSetting, obj.ServiceName, storedHosts, insecureEdgeTermAllow, obj)
+		/// TODO: Need to work from here.
+		if lib.CheckObjectNameLength(poolNode.Name, lib.Pool) {
+			isPoolNameLenExceedAviLimit = true
+		}
 		if !lib.GetNoPGForSNI() || !isIngr {
+			isPoolNameLenExceedAviLimit = false
 			pool_ref := fmt.Sprintf("/api/pool?name=%s", poolNode.Name)
+
 			ratio := obj.weight
-			pgNode.Members = append(pgNode.Members, &avimodels.PoolGroupMember{PoolRef: &pool_ref, Ratio: &ratio})
-			if vsNode[0].CheckPGNameNChecksum(pgNode.Name, pgNode.GetCheckSum()) {
-				vsNode[0].ReplaceSniPGInSNINode(pgNode, key)
+			if !isPoolNameLenExceedAviLimit {
```

Comment:
This condition will always be true. Can you explain the logic?

Reply:
This is the miss. I will correct it.

Reply:
addressed